src/t_set.c

```diff
@@ -184,13 +184,15 @@ int setTypeAddAux(robj *set, char *str, size_t len, int64_t llval, int str_is_sd
             {
                 /* In the "safe to add" check above we assumed all elements in
                  * the intset are of size maxelelen. This is an upper bound. */
-                setTypeConvert(set, OBJ_ENCODING_LISTPACK);
+                setTypeConvertAndExpand(set, OBJ_ENCODING_LISTPACK,
+                                        intsetLen(set->ptr) + 1, 1);
```

Comment:
Not sure if we need to convert to listpack like dict, the overhead of converting listpack is small.

Reply:
When we have this implementation, we can as well use it here? (even if the estimated size isn't very accurate for listpack)